modules/developer_manual/pages/app/fundamentals/database.adoc

```diff
@@ -42,6 +42,29 @@ class AuthorDAO {
 }
 ----
 
+Database programming guidelines
+-------------------------------
+
+* Always use the Query Builder
+* Don't update more than 1 million rows within a transaction due to DB limitations
+* Don't add more than 999 conditions in a `WHERE ... IN ...` statement but chunk it into separate queries when using SQLite
+* When processing big tables, always do this in chunks, don't store the whole table in memory
+* Oracle compatibility specifics:
+** For Oracle `null` and empty strings are the same thing. Need special handling to catch these cases
+** When reading values, make sure to convert null to empty string when expected
+** Keep this in mind when designing DB schema and values expectations.
+** When using a condition based on empty strings, use `is not null` with Oracle instead
+** Oracle can only compare the first 4000 bytes of a `CLOB` column
+** Make sure to properly escape column names when using custom functions with `createFunction`. the escaping is usually done automatically by the query builder otherwise. Oracle is the most likely to complain about unquoted columns while other databases will work fine
```

Comment:
s/the escaping/The escaping/g
"otherwise", at the end of the second sentence seems superfluous.